clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/security/KubernetesNamedAccountCredentials.java

```diff
@@ -89,8 +89,12 @@ public KubernetesNamedAccountCredentials(
     return requiredGroupMembership;
   }
 
+  /**
+   * Get the namespaces without making a call to the kubernetes cluster. If the cache is empty,
+   * return an empty list.
+   */
   public List<String> getNamespaces() {
-    return credentials.getDeclaredNamespaces();
```

Comment:
> It doesn't seem the method `KubernetesNamedAccountCredentials.getNamespaces` is called by anyone

I agree...grepping through the code comes up empty.  However, [this code in the `/credentials` endpoint](https://github.com/spinnaker/clouddriver/blob/4e9a3994cac536a91a0f907781f5eb56aedeef46/clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/CredentialsController.groovy#L67) ends up calling the method as part of serialization.

> The `KubernetesCredentialsLifecycleHandler` class makes this call:
> 
> ```
> // Attempt to get namespaces to resolve any connectivity error without blocking /credentials
> List<String> namespaces = credentials.getCredentials().getDeclaredNamespaces();
> ```
> 
> So it's invoking directly the method in `KubernetesCredentials`.

Looks like it makes sense to change the call there, to match the comment.  I don't think I've run into that problem though.  I'm happy to make that change in this PR though if you like.

Reply:
Oh that serialization was a surprise. `namespaces` is only returned if the parameter `expand=true` is included in the request to `/credentials`, but since `CredentialsController` is not specific to kubernetes we cannot add logic there to call `getDeclaredNamespacesFromCache` or `getDeclaredNamespaces`.

I think what you have is ok to not block `/credentials` call even if details are not requested, and I noticed the call in `KubernetesCredentialsLifecycleHandler` only happens after an account is updated (not when it's added), so it's ok to have the live call there.

So my only comment for this PR would be to log a warning when `getDeclaredNamespacesFromCache` is called and it returns an empty list because there is no cache yet.

Reply:
https://github.com/spinnaker/clouddriver/pull/5363/commits/5751e7c66df7c04fcbeede66e8f1c7f309ab808c